### PR TITLE
Added codeclimate workflow to Actions

### PR DESCRIPTION
**Pull request by neontvn**

### CODE REVIEW

**Comment by Kshitijde** [comment]
Older dependancy being used right now, confirm on whether an updated version is available

File: specs/adrs/01_code_coverage_tool.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+The implementation will be confirmed through code reviews and a CI pipeline that ensures the coverage report is generated in the required formats and is published to CodeClimate and that it runs efficiently with the current test suite. 
+
+# Pros and Cons of the Options
+
+C8
+
+- Good, because it has a simpler setup for TypeScript and Mocha
+- Good, because it is fast and optimized for performance
+- Good, because it produces essential coverage reports without extra configuration
+- Bad, because it is a newer tool with a smaller community
```
Comment (DylanLukes) [inline specs/adrs/01_code_coverage_tool.md:59]: This seems like the main point of concern. However, it seems like C8 is pretty small and sticks to existing standard formats for input/output. So, it should be easy to get out of if you need to switch.

Comment (DylanLukes) [reply specs/adrs/01_code_coverage_tool.md:59]: One other small note: considering the ADRs as historical records it is good to be a little more specific. For example, note the (at time of writing) activity/stars/etc on GitHub, first release, etc. Very minor note and more for in the real world though, you can just skip over it here.

**Review by DylanLukes — APPROVED** [review]
This looks good to me. I haven't actually used either `nyc` or `c8`, but from your ADR, it seems like your choice is well reasoned.

File: specs/adrs/01_code_coverage_tool.md (hunk 1 above)
Comment (neontvn) [reply specs/adrs/01_code_coverage_tool.md:59]: Sure, noted. Thank you for the input.

Comment (Kshitijde) [reply specs/adrs/01_code_coverage_tool.md:59]: Thanks Dylan, will keep that in mind

**Review by Kshitijde — APPROVED** [review]

**Pull request merged**